spec/mongo/client_construction_spec.rb

```diff
@@ -55,6 +58,37 @@
         end
       end
     end
+
+    context 'with scan: false' do
+      it 'does not perform i/o' do
+        allow_any_instance_of(Mongo::Server::Monitor).to receive(:run!)
+        expect_any_instance_of(Mongo::Server::Monitor).not_to receive(:scan!)
+        start_time = Time.now
+        # return should be instant
+        c = Timeout.timeout(1) do
+          ClientRegistry.instance.new_local_client(['1.1.1.1'], scan: false)
+        end
+        expect(c.cluster.servers).to be_empty
+        c.close
+      end
+    end
+
+    context 'with default scan: true' do
+      # TODO this test requires there being no outstanding background
+      # monitoring threads running, as otherwise the scan! expectation
+      # can be executed on a thread that belongs to one of the global
+      # clients for instance
+      it 'performs one round of sdam' do
+        # Does not work due to
+        # https://github.com/rspec/rspec-mocks/issues/1242.
+        #expect_any_instance_of(Mongo::Server::Monitor).to receive(:scan!).
```

Comment:
It looks like the issue is closed and they don't have any intention of fixing it. Given that it doesn't seem like there's any expectation of being able to use this, can we delete it?

Reply:
If I keep running into this I will look into fixing it myself.

Reply:
Sounds good